### PR TITLE
Installing Schemaspy

### PR DESCRIPTION
**Pull request by poggdogg**
As a technical team I want to be able to review the data model that PRIME is using with a tool like schemaspy.

### CODE REVIEW

File: schemaspy/pgsql-oracle-fdw.properties [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Current State
+# The following configuration allows SchemaSpy to query and graph the Foreign tables imported by oracle-fdw.
```
Comment (itchison) [inline schemaspy/pgsql-oracle-fdw.properties]: We don't need this.. and also I see some files in /schemaspy/conf that are duplicates of these files. Please review.

File: schemaspy/conf/pgsql-oracle-fdw.properties [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Current State
```
[the remark this answers is not in the document]
Comment (poggdogg) [reply schemaspy/conf/pgsql-oracle-fdw.properties]: yep

File: schemaspy/Dockerfile.oracle [hunk 3]
```diff
@@ -0,0 +1,27 @@
+# Assumes you have built the base image and called it `schema-spy`
```
[the remark this answers is not in the document]
Comment (poggdogg) [reply schemaspy/Dockerfile.oracle]: No, more legacy commentary from the old code

File: schemaspy/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        wget \
+        ca-certificates \
+        librsvg \
+        graphviz \
+        ttf-ubuntu-font-family && \
+    mkdir lib && \
+    wget -nv -O lib/schemaspy-$SCHEMA_SPY_VERSION.jar https://github.com/schemaspy/schemaspy/releases/download/v$SCHEMA_SPY_VERSION/schemaspy-$SCHEMA_SPY_VERSION.jar && \
+    cp lib/schemaspy-$SCHEMA_SPY_VERSION.jar lib/schemaspy.jar && \
+    wget -nv -O lib/pgsql-jdbc.jar http://central.maven.org/maven2/org/postgresql/postgresql/$POSTGRESQL_VERSION/postgresql-$POSTGRESQL_VERSION.jar && \
+    wget -nv -O lib/mysql-jdbc.jar http://central.maven.org/maven2/mysql/mysql-connector-java/$MYSQL_VERSION/mysql-connector-java-$MYSQL_VERSION.jar && \
```
[the remark this answers is not in the document]
Comment (poggdogg) [reply schemaspy/Dockerfile]: Yeah, this was built off of the Pathfinder system  I'll remark those out and give it a try.

File: schemaspy/openshift.dockerfile [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+ENV LC_ALL C
+
+# Define the default output directory for SchemaSpy
+# If you change this you will need to update the Caddy configuration.
+ENV OUTPUT_PATH=/var/www/html
+
+# Define the default versions for the image
+ENV SCHEMA_SPY_VERSION=6.1.0
+ENV POSTGRESQL_VERSION=42.2.1
+ENV MYSQL_VERSION=6.0.6
```
[the remark this answers is not in the document]
Comment (poggdogg) [reply schemaspy/openshift.dockerfile]: yep

File: schemaspy/pgsql-oracle-fdw.properties (hunk 1 above)
Comment (poggdogg) [reply schemaspy/pgsql-oracle-fdw.properties]: yep, I'll look into the dupes

File: openshift/schemaspy.dc.yaml [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+    required: false
+  - name: "DATABASE_HOST"
+    displayName: "The host name (or ip address) for the database server; override to match your environment.  In the case of Oracle databases this must be a <hostname|ipaddress>:<port> pair even if using the default Oracle port; a requirement of the Oracle Thin SchemaSpy process."
+    required: true
+    value: "postgresql"
+  - name: "DATABASE_NAME"
+    displayName: "Database Name"
+    required: true
+    description: "Name of the deployed database (see environment variables in deployed database)."
+    value: "TheOrgBook_Database"
```
Comment (itchison) [inline openshift/schemaspy.dc.yaml]: Copy paste issue?

File: schemaspy/conf/pgsql-oracle-fdw.properties (hunk 2 above)
[the remark this answers is not in the document]
Comment (itchison) [reply schemaspy/conf/pgsql-oracle-fdw.properties]: looks like it's still here.

File: schemaspy/conf/sqlite.properties [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Alternate properties file for sqlite - the built-in one is out of date
```
[the remark this answers is not in the document]
Comment (itchison) [reply schemaspy/conf/sqlite.properties]: I don't think this file is needed.

File: openshift/schemaspy.dc.yaml (hunk 6 above)
Comment (poggdogg) [reply openshift/schemaspy.dc.yaml]: yeah, updating.

**Pull request merged**